From f097a7c53e0b8ca2241cd0132fa338aafdcfdaa3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:37 -0400
Subject: [tools] perf/script: Move evname print code to process_event()

Message-id: <1408953210-26343-30-git-send-email-jolsa@redhat.com>
Patchwork-id: 88743
O-Subject: [PATCH RHEL7.1 BZ1133083 029/322] perf script: Move evname print code to process_event()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit e944d3d7d151eea149c62310eaff7b92c7732f58
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Nov 18 14:34:52 2013 +0900

description
===========
The print_sample_start() will be reused by other printing routine for
internal events like COMM, FORK and EXIT from next patch.  And because
they're not tied to a specific event, move the evname print code to its
caller.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-script.c b/tools/perf/builtin-script.c
index baf1798..b392770 100644
--- a/tools/perf/builtin-script.c
+++ b/tools/perf/builtin-script.c
@@ -288,7 +288,6 @@ static void print_sample_start(struct perf_sample *sample,
           struct perf_evsel *evsel)
 {
  struct perf_event_attr *attr = &evsel->attr;
- const char *evname = NULL;
  unsigned long secs;
  unsigned long usecs;
  unsigned long long nsecs;
@@ -323,11 +322,6 @@ static void print_sample_start(struct perf_sample *sample,
   usecs = nsecs / NSECS_PER_USEC;
   printf("%5lu.%06lu: ", secs, usecs);
  }
-
- if (PRINT_FIELD(EVNAME)) {
-  evname = perf_evsel__name(evsel);
-  printf("%s: ", evname ? evname : "[unknown]");
- }
 }
 
 static bool is_bts_event(struct perf_event_attr *attr)
@@ -434,6 +428,11 @@ static void process_event(union perf_event *event, struct perf_sample *sample,
 
  print_sample_start(sample, thread, evsel);
 
+ if (PRINT_FIELD(EVNAME)) {
+  const char *evname = perf_evsel__name(evsel);
+  printf("%s: ", evname ? evname : "[unknown]");
+ }
+
  if (is_bts_event(attr)) {
   print_sample_bts(event, sample, evsel, machine, thread);
   return;
-- 
1.7.1